src/raw.rs

```diff
@@ -232,6 +255,8 @@ where
     unsafe fn wake(ptr: *const ()) {
         // This is just an optimization. If the schedule function has captured variables, then
         // we'll do less reference counting if we wake the waker by reference and then drop it.
+        // TODO: Add safety docs here. What requirements does ptr have to meet? We can probably
+        // assume that it has to be a pointer to a properly-allocated task.
```

Comment:
For all of the `RawTask` functions it is assumed `ptr` points to a valid task. This is enforced as the VTable functions are only called with the `ptr` from the `alloc` function.

Reply:
Added comment here.